spring/boot-starter/src/main/java/org/apache/shardingsphere/elasticjob/spring/boot/tracing/ElasticJobTracingConfiguration.java

```diff
@@ -68,9 +67,11 @@ public DataSource tracingDataSource(final TracingProperties tracingProperties) {
          * @return a bean of tracing configuration
          */
         @Bean
-        @ConditionalOnBean(DataSource.class)
-        public TracingConfiguration<DataSource> tracingConfiguration(@Qualifier("dataSource") final DataSource dataSource,
+        public TracingConfiguration<DataSource> tracingConfiguration(@Qualifier("dataSource") @Nullable final DataSource dataSource,
                                                                      @Qualifier("tracingDataSource") @Nullable final DataSource tracingDataSource) {
+            if (dataSource == null && tracingDataSource == null) {
+                throw new IllegalArgumentException("Neither dataSource nor tracingDataSource bean is required");
+            }
```

Comment:
- I would say this is a bit illogical. In fact, `org.apache.shardingsphere.elasticjob:elasticjob-spring-boot-starter` should not have a mandatory dependency on `org.springframework.boot:spring-boot-starter-jdbc`.
- I may raise another PR later to try to remove the mandatory dependency on `org.springframework.boot:spring-boot-starter-jdbc`. Or are you going to do that? Because the Spring Bean for `TracingConfiguration` should actually only be registered when `elasticjob.tracing.type` is `RDB`.

Reply:
The dependency `org.springframework.boot:spring-boot-starter-jdbc` is optional. It attempts to retrieve the DataSource from `elasticjob.tracing.dataSource`. If it is not found, it will then obtain the default DataSource from context.

Reply:
- I missed the context of the current class earlier. The current PR looks fine. I need a little time to see why CI is failing with unstable.